vulnerabilities/importers/pysec.py

```diff
@@ -0,0 +1,178 @@
+        2. unzip the file
+        3. open the file one by one
+        4. yield the json file to parse_advisory_data
+        """
+        url = "https://osv-vulnerabilities.storage.googleapis.com/PyPI/all.zip"
+        response = requests.get(url).content
+        try:
+            with ZipFile(BytesIO(response)) as zip_file:
+                for file_name in zip_file.namelist():
+                    with zip_file.open(file_name) as f:
+                        vul_info = json.loads(f.read())
+                        yield parse_advisory_data(vul_info)
+        except requests.exceptions.RequestException:
+            logger.error(f"Failed to fetch osv-vulnerabilities PyPI: HTTP ")
+
+
+def parse_advisory_data(raw_data: dict) -> AdvisoryData:
+    if "summary" in raw_data:
+        summary = raw_data["summary"]
+    else:
+        summary = ""
+
+    aliases = raw_data["aliases"] if "aliases" in raw_data else raw_data["id"]
+
+    if "published" in raw_data:
+        date_published = datetime.strptime(
+            raw_data["published"][0:19], "%Y-%m-%dT%H:%M:%S"
+        ).replace(tzinfo=timezone.utc)
+    else:
+        date_published = None
+        logger.error("date_published not found " + raw_data["id"])
+
+    severity = []
+    if "severity" in raw_data:
+        for sever_list in raw_data["severity"]:
+            if "type" in sever_list and sever_list["type"] == "CVSS_V3":
+                severity.append(
+                    VulnerabilitySeverity(
+                        system=SCORING_SYSTEMS["cvssv3_vector"],
+                        value=sever_list["score"],
+                    )
+                )
+    elif "ecosystem_specific" in raw_data and "severity" in raw_data["ecosystem_specific"]:
+        severity.append(
+            VulnerabilitySeverity(
+                system=SCORING_SYSTEMS["generic_textual"],
+                value=raw_data["ecosystem_specific"]["severity"],
+            )
+        )
+    else:
+        severity = []
+        logger.warning(f"severity not found- {raw_data['id']!r}")
+
+    references = []
+    if "references" in raw_data:
+        references = [
+            Reference(url=ref["url"], severities=severity) for ref in raw_data["references"] if ref
+        ]
+
+    affected_package = []
+    if "affected" in raw_data:
+        for affected_pkg in raw_data["affected"]:
+            package = affected_pkg["package"]
+            if "ecosystem" in package and "name" in package:
+                purl = PackageURL(type=package["ecosystem"], name=package["name"])
+            else:
+                purl = ""
+                logger.error(f"purl affected_pkg not found - {raw_data['id'] !r}")
+
+            affected_version_range = None
+            if "versions" in affected_pkg:
+                affected_version_range = (
+                    # HINT:  Values larger than 1/3 of a buffer page cannot be indexed.
+                    PypiVersionRange(affected_pkg["versions"][slice(200)])
```

Comment:
I use `slice()` to get ride of this error , Any Suggestion ?  
![Screenshot from 2022-03-08 18-30-52](https://user-images.githubusercontent.com/29133904/157668519-1ea010cc-9601-486e-b1f4-c12d9319c416.png)

Reply:
django.db.utils.OperationalError: index row size 2720 exceeds btree version 4 maximum 2704 for index "vulnerabilities_advisory_aliases_summary_affected_d8f3d9e1_uniq"
DETAIL:  Index row references tuple (11064,3) in relation "vulnerabilities_advisory".
HINT:  Values larger than 1/3 of a buffer page cannot be indexed.
Consider a function index of an MD5 hash of the value, or use full text indexing.

Reply:
Thanks!... so we have a model problem then... 

Which record does trigger this issue? which upstream advisory URL has this?

The solution suggested by PostgreSQL may be OK:

`Consider a function index of an MD5 hash of the value, or use full text indexing.`

My guess or my recall is that there is a very long list of enumerated affected versions...If this makes the DB choke as we have an index on this, there multiple solutions alright. All would involved changing the unique together and not use the full field details as a text blob.

1. add a new field for a checksum (or some DB index/constraint function without a field?) that would be used for the unique together and would always be small. This needs good design as it requires recomputing when the value is updated... unless we would never update this imported data field. 
2. change the design to create a new model to store these in a table. This may still require 1. but the data would never exceed any size limit of a text field. This may be the right time 
3. treat this as a one time exception: my guess is that this is a long enumeration of ALL the versions of a package up to a point in time. This could be instead treated as a range with a one-off hardcoded hack.
4. do nothing just now: just log the problem  and enter an issue to revisit this  

Not sure which one is best. In anycase, we never want to store truncated data.